### PR TITLE
[28] Add methods to convert markdown to json

### PR DESCRIPTION
**Pull request by bobbylej**
- Add utils with methods to convert markdown from text to json and from json to jsx
- Change names for files with utils

### CODE REVIEW

File: src/app/shared/components/AsideGraphicEditor/AsideGraphicEditor.tsx [hunk 1]
```diff
@@ -2,7 +2,7 @@ import React, { ReactElement } from 'react';
 import { Grid, Typography } from '@material-ui/core';
 import { Circle } from 'app/shared/components/Circle/Circle';
 import { useAsideGraphicEditor } from 'app/shared/components/AsideGraphicEditor/AsideGraphicEditor.styles';
-import { AsideGraphic } from 'app/shared/interfaces/asideGraphic';
+import { AsideGraphic } from 'app/shared/interfaces/asideGraphic.interface';
```
Comment (lukaszzadamczyk) [inline src/app/shared/components/AsideGraphicEditor/AsideGraphicEditor.tsx:6]: do we have to do that?😏

File: src/app/shared/components/Circle/Circle.styles.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import { Theme } from '@material-ui/core';
 import { makeStyles } from '@material-ui/styles';
-import { pxToRem } from 'app/shared/utils/stylesUtils';
+import { pxToRem } from 'app/shared/utils/styles.utils';
```
Comment (lukaszzadamczyk) [inline src/app/shared/components/Circle/Circle.styles.ts:4]: also here? 😁

File: src/app/shared/utils/markdownJsonToJsx.utils.tsx [hunk 3]
```diff
@@ -0,0 +1,364 @@
...
+
+export const getMarkdownRuleJsxConverter = (
+  markdownRuleType: MarkdownRuleType,
+): ((
+  markdownRuleElementJson: MarkdownRuleElementJson,
+  index: number,
+  parentKey?: string,
+  props?: MarkdownRuleProps,
+) => React.ReactElement) => {
+  switch (markdownRuleType) {
```
Comment (lukaszzadamczyk) [inline src/app/shared/utils/markdownJsonToJsx.utils.tsx:55]: twice the arrow function? 🤓

**Review by lukaszzadamczyk — APPROVED** [review]

File: src/app/shared/utils/markdownJsonToJsx.utils.tsx (hunk 3 above)
Comment (bobbylej) [reply src/app/shared/utils/markdownJsonToJsx.utils.tsx:55]: first one is to define type of the function which returns another function 😄

File: src/app/shared/components/AsideGraphicEditor/AsideGraphicEditor.tsx (hunk 1 above)
Comment (bobbylej) [reply src/app/shared/components/AsideGraphicEditor/AsideGraphicEditor.tsx:6]: yup, let's keep such convention 😉

File: src/app/shared/components/Circle/Circle.styles.ts (hunk 2 above)
Comment (bobbylej) [reply src/app/shared/components/Circle/Circle.styles.ts:4]: yup, let's keep such convention 😉

**Pull request merged**